src/uct/ib/base/ib_device.h

```diff
@@ -52,8 +52,9 @@ enum {
 enum {
     UCT_IB_DEVICE_FLAG_MLX4_PRM = UCS_BIT(1),   /* Device supports mlx4 PRM */
     UCT_IB_DEVICE_FLAG_MLX5_PRM = UCS_BIT(2),   /* Device supports mlx5 PRM */
-    UCT_IB_DEVICE_FLAG_DC       = UCS_BIT(3),   /* Device supports DC */
-    UCT_IB_DEVICE_FLAG_LINK_IB  = UCS_BIT(4)    /* Require only IB */
+    UCT_IB_DEVICE_FLAG_MELLANOX = UCS_BIT(3),   /* Mellanox device */
```

Comment:
Why do you need a separate flag/bit when you already have MLX?_PRM.  
Do  UCT_IB_DEVICE_FLAG_MLX5_PRM & UCT_IB_DEVICE_FLAG_MLX4_PRM

Reply:
in order to know if it's ok to check FW version

Reply:
Do you have devices that will have UCT_IB_DEVICE_FLAG_MELLANOX but not (UCT_IB_DEVICE_FLAG_MLX5_PRM or UCT_IB_DEVICE_FLAG_MLX4_PRM) ?

Reply:
probably not in near future, but i wanted to separate the vendor flag and the "PRM" flag. I can also check MLX4|MLX5 if you think it's cleaner.

Reply:
than you can define UCT_IB_DEVICE_FLAG_MELLANOX = MLX4|MLX5 ?
But ultimately it is up to you. I just want to avoid confusion with all the flags.